sdks/python/apache_beam/io/gcp/gcsio.py

```diff
@@ -296,160 +234,87 @@ def delete_batch(self, paths):
     """
     if not paths:
       return []
-
-    paths = iter(paths)
+    if len(paths) > MAX_BATCH_OPERATION_SIZE:
+      raise TooManyRequests("Batch larger than %s", MAX_BATCH_OPERATION_SIZE)
     result_statuses = []
-    while True:
-      paths_chunk = list(islice(paths, MAX_BATCH_OPERATION_SIZE))
```

Comment:
We may still need `MAX_BATCH_OPERATION_SIZE` logic as the documentation states "No more than 100 calls should be included in a single batch request.": https://cloud.google.com/storage/docs/batch#batch-example-request

Reply:
The lines I added above should ensure that an error is raised if more than 100 paths are requested. I'm pretty sure the previous logic was just truncating the request. If that's what we want to do, I can change it back, but I think throwing an error is more accurate.

Reply:
Should not raise error here. Deleting more than 100 objects was supported before.

Reply:
The previous code was silently truncating the list if more than 100 paths were given. Is that preferable to throwing an error so folks know they're over the limit?

Reply:
I see you are write for the previous behavior, and it should be considered a bug (after 100th element ignored). This might be a source of leftover temp test files in our test system. How about divide the incoming files into slices of 100, and sending batch request sequentially?

```
for i in range(0, len(paths), MAX_BATCH_OPERATION_SIZE):
  slice = paths[i:i + MAX_BATCH_OPERATION_SIZE]
  try:
    with self.client.batch():
      for path in paths::
      bucket_name, blob_path = parse_gcs_path(path)
      bucket = self.client.get_bucket(bucket_name)
      blob = storage.Blob(blob_path, bucket)
      blob.delete()
  except Exception as err:
    ...
```

btw max batch size is 1000 for storage client:  https://github.com/googleapis/python-storage/blob/2b449cd289cb573ca1653bba37ecb84d35a025ad/google/cloud/storage/batch.py#LL138C5-L138C21)

Looking into the client library source code, it only tracks the lastly saw error (and drops previous one): https://github.com/googleapis/python-storage/blob/2b449cd289cb573ca1653bba37ecb84d35a025ad/google/cloud/storage/batch.py#LL240C19-L240C19 So we would not able to get the status of single delete requests as it is for now.

Reply:
Okay, I'll change our MAX_BATCH_OPERATION_SIZE to 1000 the GCS client and see if I can make slicing larger groups of files in multiple batches work, because I do agree that would be better.

Re: that link the client's code, I think that's outdated. There's a parameter now to record a different exception for each request. I'm in the process of implementing it currently to ensure that `delete_batch` is idempotent.

Reply:
Oh wait, I'm now seeing that the GCS client feature I mentioned is actually very new and not yet part of a release. I've reached out the client team to see when a new release might be available, since it will could really help solve our issue with making delete_batch idempotent.